File: GildedRose/src/com/gildedrose/items/AgedItem.java
package com.gildedrose.items;

public class AgedItem extends ItemDecorator {
	
	// Constructors
	protected AgedItem(Item itemToDecorate)
	{
		super(itemToDecorate);
	}

	// Protected methods
	@Override
	protected void updateQuality()
	{
		if (this.isSellInEnded())
		{
			this.increaseQualityTwice();
		}
		else
		{
			this.increaseQualityOnce();
		}
	}

}

File: GildedRose.Net/Items/AgedItem.cs
namespace GildedRose.Items
{
    public class AgedItem : ItemDecorator
    {
    
        internal AgedItem(Item itemToDecorate) : base(itemToDecorate) { }
        
        // Protected methods
        protected override void UpdateQuality()
        {
            if (this.SellInEnded)
            {
                this.IncreaseQualityTwice();
            }
            else
            {
                this.IncreaseQualityOnce();
            }
        }
        
    }
}